From 1affeeffef1a0122fda9917b1d252b5f93d75eef Mon Sep 17 00:00:00 2001
From: Geetha sowjanya <gakula@marvell.com>
Date: Wed, 18 Sep 2019 16:07:29 +0530
Subject: [PATCH 349/767] octeontx2-pf: Add shutdown sequence for PF/VF

commit 90c64b10f51d921844ada550493594877724686a from
git@git.assembla.com:cavium/WindRiver.linux.git

This patch adds pci driver shutdown support device
teardown on system reboot.

Change-Id: I820ee817c1b8ed399d3e309b5060334c03c5f5e5
Signed-off-by: Geetha sowjanya <gakula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/16060
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 1 +
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 0e4e0bdd6580..1a0f08be5a33 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -2423,6 +2423,7 @@ static struct pci_driver otx2_pf_driver = {
 	.name = DRV_NAME,
 	.id_table = otx2_pf_id_table,
 	.probe = otx2_probe,
+	.shutdown = otx2_remove,
 	.remove = otx2_remove,
 	.sriov_configure = otx2_sriov_configure
 };
diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
index efdaacff36fb..777085f613d2 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_vf.c
@@ -631,6 +631,7 @@ static struct pci_driver otx2vf_driver = {
 	.id_table = otx2_vf_id_table,
 	.probe = otx2vf_probe,
 	.remove = otx2vf_remove,
+	.shutdown = otx2vf_remove,
 };
 
 static int __init otx2vf_init_module(void)
-- 
2.31.1

